src/main/scala/org/hammerlab/guacamole/readsets/rdd/PartitionedRegions.scala

```diff
@@ -1,98 +1,214 @@
 package org.hammerlab.guacamole.readsets.rdd
 
-import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.rdd.RDD
-import org.hammerlab.guacamole.distributed.{KeyPartitioner, PileupFlatMapUtils, TaskPosition, WindowFlatMapUtils}
-import org.hammerlab.guacamole.loci.partitioning.LociPartitioning
-import org.hammerlab.guacamole.logging.DelayedMessages
+import org.apache.spark.{Accumulable, SparkContext}
+import org.hammerlab.guacamole.loci.partitioning.LociPartitioner.PartitionIndex
+import org.hammerlab.guacamole.loci.partitioning.{LociPartitionerArgs, LociPartitioning}
+import org.hammerlab.guacamole.loci.set.LociSet
 import org.hammerlab.guacamole.logging.LoggingUtils.progress
-import org.hammerlab.guacamole.readsets.PerSample
+import org.hammerlab.guacamole.readsets.{NumSamples, PerSample, SampleId}
 import org.hammerlab.guacamole.reference.ReferenceRegion
-import org.hammerlab.guacamole.windowing.SlidingWindow
+import org.hammerlab.magic.accumulables.{HistogramParam, HashMap => MagicHashMap}
+import org.hammerlab.magic.rdd.KeyPartitioner
+import org.hammerlab.magic.stats.Stats
 
 import scala.reflect.ClassTag
 
 /**
- * This object encapsulates functionality for partitioning reference-mapped regions for processing as "pileups".
+ * Groups a [[LociPartitioning]] with an [[RDD[ReferenceRegion]]] that has already been partitioned according to that
+ * partitioning.
  *
- * Given an RDD of regions, it:
- *
- *   - partitions them by genomic loci,
- *   - makes multiple copies of regions that straddle partition boundaries.
- *
- * The result is that, given knowledge of which ranges of genomic loci correspond to which partition of the resulting
- * RDD, downstream code can iterate over the partitions emitted here and build full pileups for the loci that each
- * partition corresponds to.
- *
- * [[WindowFlatMapUtils]] calls this, and uses [[SlidingWindow]] to iterate over the partitioned regions in a manner
- * suitable for pileup-construction by [[PileupFlatMapUtils]].
+ * This means some regions will occur multiple times in the RDD (due to regions straddling partition boundaries, so it's
+ * important not to confuse this with a regular RDD[ReferenceRegion].
  */
+class PartitionedRegions[R <: ReferenceRegion: ClassTag](val numSamples: NumSamples,
+                                                         @transient keyedRegions: RDD[(SampleId, R)],
```

Comment:
I don't think it's necessary, but defensive `@transient`ing seems like reasonable practice, since e.g. spark shells will pick up and serialize any variables that are in scope if a closure defined in the shell is passed to RDD.map or equivalent.

I'll add a comment saying as much.